File: TripKitUI/views/TKUIStyledLabel.m
//
//  TKUIStyledLabel.m
//  TripKit
//
//  Created by Kuan Lun Huang on 30/03/2015.
//
//

#import "TKUIStyledLabel.h"

#ifdef TK_NO_MODULE
#import "TripKit.h"
#import "TripKit/TripKit-Swift.h"
#else
@import TripKit;
#import "TripKitUI/TripKitUI-Swift.h"
#endif

#import "TKStyleManager+TripKitUI.h"

@implementation TKUIStyledLabel

#pragma mark - Initializers

- (instancetype)init
{
  self = [super init];
  
  if (self) {
    [self commonInit];
  }
  
  return self;
}

- (instancetype)initWithFrame:(CGRect)frame
{
  self = [super initWithFrame:frame];
  
  if (self) {
    [self commonInit];
  }
  
  return self;
}

- (id)initWithCoder:(NSCoder *)aDecoder
{
  self = [super initWithCoder:aDecoder];
  
  if (self) {
    [self commonInit];
  }
  
  return self;
}

#pragma mark - Private methods

- (void)commonInit
{
  NSString *style = [[self.font fontDescriptor] fontAttributes][UIFontDescriptorTextStyleAttribute];
  
  if (! style) {
    CGFloat point = self.font.pointSize;
    self.font = [TKStyleManager systemFontWithSize:point];
  } else {
    self.font = [TKStyleManager customFontForTextStyle:style];
  }
}
  

@end

File: Sources/TripKitUI/views/TKUIStyledLabel.swift
//
//  TKUIStyledLabel.swift
//  TripKitUI-iOS
//
//  Created by Adrian Schönig on 31/1/21.
//  Copyright © 2021 SkedGo Pty Ltd. All rights reserved.
//

import UIKit
import TripKit

@objc
open class TKUIStyledLabel: UILabel {
  public override init(frame: CGRect) {
    super.init(frame: frame)
    didInit()
  }
  
  public required init?(coder: NSCoder) {
    super.init(coder: coder)
    didInit()
  }
  
  private func didInit() {
    if let style = font.fontDescriptor.fontAttributes[.textStyle] as? String {
      font = TKStyleManager.customFont(forTextStyle: UIFont.TextStyle(rawValue: style))
    } else {
      font = TKStyleManager.systemFont(size: font.pointSize)
    }
  }
}
